test/Utils/System.hpp

```diff
@@ -85,6 +87,64 @@ launchTarantool(bool enable_ssl = false)
 	exit(EXIT_FAILURE);
 }
 
+#else
+
+/**
+ * Launches intermediate process, connected with parent by pipe.
+ * The intermediate process lauches another process, running tarantool, and then
+ * falls asleep on reading from pipe. When parent process is dead, the pipe
+ * will be closed, and the intermediate process will read EOF. Right after, it
+ * will kill its child process, which is Tarantool, and exit.
+ * In the case, when the intermediate process fails to fork Tarantool, it kills
+ * the parent process and terminates.
+ */
```

Comment:
I agree with @alyapunov in the sense that this is code is really sophistacted, but I guess there is no other way to implement this approach on macOS, so I see two ways to make life easier for us:
1. Since the approach @drewdzzz proposes here is portable, let's use it everywhere: i.e., drop the Linux version and only leave this one.
2. Abandon this approach entirely, and instead refactor our test code to save the Tarantool PID (perhaps, create some `TarantoolInstance` object), and clean it up properly when testing fails. An appraoch like this is used in go-tarantool, for instance [^1]. Not sure what to do, if the test crashes though (seems like the process leaks in this case), I guess we should consult with someone from the @tarantool/ecosystem, like @DifferentialOrange or @0x501D.

[^1]: https://github.com/tarantool/go-tarantool/blob/bd6aab9db09260abe3bfe929b23e81a0171abf2b/shutdown_test.go#L137-L139

Reply:
> Not sure what to do, if the test crashes though

go-tarantool is also unable to shutdown Tarantool processes in case of a crash now, so this problem is unresolved for us as well.

Reply:
> > Not sure what to do, if the test crashes though
> 
> go-tarantool is also unable to shutdown Tarantool processes in case of a crash now, so this problem is unresolved for us as well.

This is an implementation flaw. It could be implemented in Go for `panic` in fact:
https://github.com/tarantool/go-tarantool/issues/147#issuecomment-1172426780

But there is no solution to the problem in go-tarantool that would be suitable for C++.

Reply:
Why do you think that a process leak happens?
I crashed the test and all processes have been deleted.

Reply:
> Why do you think that a process leak happens?

I mean if we go for option 2.

Reply:
In this case, `kill` is signal-safe, so we could kill Tarantool from signal handler.

Reply:
> In this case, `kill` is signal-safe, so we could kill Tarantool from signal handler.

Yeah, I also thought about this, we could add signal handlers from recoverable signals (i.e., even go can't do anything about `SIGKILL`).

Another path we can take is setting up and cleaning up Tarantool externally (i.e., via the test harness), but it may not be scalable or convenient.

Reply:
I guess it's time for @alyapunov to take the decision, how we should proceed.

Reply:
I guess it's time for @alyapunov to take the decision how we should proceed here.